qutebrowser/mainwindow/statusbar/backforward.py

```diff
@@ -37,6 +37,8 @@ def on_tab_cur_url_changed(self, tabs):
 
     def on_tab_changed(self, tab):
         """Update the text based on the given tab."""
+        if self.isHidden():
+            return
```

Comment:
You should probably still set the text in that case, but not make it visible. Right now, when removing the widget and re-adding it on another tab, it shows the wrong text.